src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,42 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        BulldozerProducer bulldozerProducer = new BulldozerProducer();
+        ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+        TruckProducer truckProducer = new TruckProducer();
+        List<? extends Machine> machines = new ArrayList<>();
+        if (type == Bulldozer.class) {
+            machines = bulldozerProducer.get();
+        } else if (type == Excavator.class) {
+            machines = excavatorProducer.get();
+        } else if (type == Truck.class) {
+            machines = truckProducer.get();
+        }
+        return new ArrayList<>(machines);
+    }
```

Comment:
we create 3 object but return just a one.
did we realy need to create other objects?